src/prototype_creep_harvest.js

```diff
@@ -43,8 +43,28 @@ Creep.prototype.handleSourcer = function() {
 };
 
 Creep.prototype.spawnCarry = function() {
-  var energyThreshold = Game.rooms[this.memory.base].controller.level * config.sourcer.spawnCarryLevelMultiplier;
-  var waitTime = config.sourcer.spawnCarryWaitTime;
+  if (this.memory.wait > 0) {
+    this.memory.wait -= 1;
+    return false;
+  }
+
+  var foundKey;
+  for (let key of Object.keys(config.carry.sizes).sort(function(a, b) { return parseInt(a, 10) - parseInt(b, 10); })) {
```

Comment:
the only posibility why we may need sort here - if someone redefined this setting in his config.local with random property order. Or if in future we'll forget about order is matter and will somehow modify setting object's propperies set after original initialisation